buildSrc/src/main/groovy/org/elasticsearch/gradle/test/ClusterConfiguration.groovy

```diff
@@ -77,6 +77,14 @@ class ClusterConfiguration {
         " " + System.getProperty('tests.jvm.argline', '')
 
     /**
+     * Should the shared environment be preserved on cluster startup? Defaults
+     * to {@code false} so we run with a clean cluster but some tests wish to
+     * preserve snapshots between clusters so they set this to true.
+     */
+    @Input
+    boolean preserveShared = false
```

Comment:
I'd been thinking of it as opting into preservation instead of out of cleaning, but I can flip, sure.